### PR TITLE
Support auth token

### PR DESCRIPTION
**Pull request by offa**
Adds support for auth token (http only).

The url format is similar to basic auth:

```
https://apitoken@host.com…
```

### CODE REVIEW

File: src/HTTP.cxx [hunk 1]
```diff
@@ -96,6 +96,11 @@ namespace influxdb::transports
         session.SetAuth(cpr::Authentication{user, pass, cpr::AuthMode::BASIC});
     }
 
+    void HTTP::setAuthToken(const std::string& token)
+    {
+        session.SetHeader(cpr::Header{{"Authorization", "Token " + token}});
```
Comment (kasparthommen) [inline src/HTTP.cxx:101]: The problem is that the header will get overwritten later on in `HTTP::send` on line 107 in the same file.

Comment (kasparthommen) [reply src/HTTP.cxx:101]: Here's a proposed solution:

https://github.dev/kasparthommen/influxdb-cxx/tree/fix-auth-token-overwriting

I have tested the code and it works.

**Review by kasparthommen — APPROVED** [review]
All good, works as expected, thanks!

**Comment by offa** [comment]
Thanks for your feedback!

**Comment by kasparthommen** [comment]
@offa  Any chance you could release a new tag on the back of this pls?

**Comment by offa** [comment]
The next release (0.8.x) has quite some work to do, but I can backport and release as a 0.7.x if that helps.

**Comment by kasparthommen** [comment]
Yes please, if it's not too much trouble!

**Comment by offa** [comment]
@kasparthommen new release is available :rocket:

**Comment by kasparthommen** [comment]
Amazing, thanks!

**Pull request merged**